test/pytest/test_softmax.py

```diff
@@ -10,50 +10,47 @@
 test_root_path = Path(__file__).parent
 
 
-def flat_distribution(shape):
-    return np.random.rand(*shape)
-
-
-def high_accuracy_distribution(shape):
-    '''Start with a flat distribution, then pick a random member of each row to amplify'''
-    x = np.random.rand(*shape)
-    imax = np.random.randint(0, shape[1], size=shape[0])
-    x[:, imax] *= 10
-    return x
+def normal_dist(shape):
+    return np.clip(np.random.normal(0, 8, shape), -32, 31)
 
 
 @pytest.fixture()
-def generate_data(function, input_shape):
-    return function((1000, *input_shape))
+def generate_data(input_shape):
+    return normal_dist((1000, *input_shape))
 
 
 @pytest.mark.parametrize('backend', ['Vivado', 'Vitis', 'Quartus'])
-@pytest.mark.parametrize('strategy', ['stable', 'argmax'])
+@pytest.mark.parametrize('strategy', ['stable', 'latency', 'argmax'])
 @pytest.mark.parametrize(
-    'function,input_shape,io_type',
+    'input_bits,input_shape,table_bits,io_type',
     [
-        (flat_distribution, (8,), 'io_parallel'),
-        (high_accuracy_distribution, (8,), 'io_parallel'),
-        (flat_distribution, (8,), 'io_stream'),
-        (high_accuracy_distribution, (8,), 'io_stream'),
-        (flat_distribution, (8, 8, 3), 'io_stream'),
-        (high_accuracy_distribution, (8, 8, 3), 'io_stream'),
+        ('16,6', (8,), '18,8', 'io_parallel'),
+        ('16,6', (8,), '18,8', 'io_stream'),
+        ('16,6', (8,), '9,6', 'io_parallel'),
+        ('16,6', (8,), '9,6', 'io_stream'),
+        ('9,6', (8,), '18,8', 'io_parallel'),
+        ('9,6', (8,), '18,8', 'io_stream'),
+        ('16,6', (8, 8, 3), '18,8', 'io_stream'),
     ],
 )
-def test_softmax(backend, strategy, generate_data, input_shape, io_type, function):
+def test_softmax(backend, strategy, generate_data, input_bits, input_shape, table_bits, io_type):
     X = generate_data
     model = tf.keras.models.Sequential()
     model.add(tf.keras.layers.Activation(input_shape=input_shape, activation='softmax', name='softmax'))
     model.compile()
 
-    f_type = 'ac_fixed<18,8,true,AC_RND,AC_SAT>' if backend == 'Quartus' else 'ap_fixed<18,8,AP_RND,AP_SAT>'
+    f_type = (
+        f'ac_fixed<{table_bits},true,AC_RND,AC_SAT>' if backend == 'Quartus' else f'ap_fixed<{table_bits},AP_RND,AP_SAT>'
+    )
     cfg = hls4ml.utils.config_from_keras_model(model, granularity='name')
     cfg['LayerName']['softmax']['Strategy'] = strategy
     cfg['LayerName']['softmax']['inv_table_t'] = f_type
     cfg['LayerName']['softmax']['exp_table_t'] = f_type
+    cfg['LayerName']['softmax_input']['Precision']['result'] = f'ap_fixed<{input_bits}>'
```

Comment:
Related to the comment above, here you're actually relying on what is essentially a bug to correctly parse even in case of Quartus.